PowerEditor/src/WinControls/OpenSaveFileDialog/CustomFileDialog.cpp

```diff
@@ -292,9 +354,10 @@ class FileDialogEventHandler : public IFileDialogEvents
 			return;
 		// Get the entered name.
 		generic_string fileName = getDialogFileName(_dialog);
+		expandEnv(fileName);
 		bool nameChanged = transformPath(fileName);
 		// Update the controls.
-		if (not ::PathIsDirectory(fileName.c_str()))
+		if (not ::PathIsDirectory(getAbsPath(fileName).c_str()))
```

Comment:
Use modern C++ op symbole `!` instead.

Reply:
Ok. It was just a bit weird with those `::`.